src/manager.cpp

```diff
@@ -160,13 +179,113 @@ void Manager::deleteSingleFruVpd(
     logging::logMessage(std::string(i_dbusObjPath));
 }
 
+bool Manager::isValidUnexpandedLocationCode(
+    const std::string& i_unexpandedLocationCode)
+{
+    if ((i_unexpandedLocationCode.length() <
+         constants::UNEXP_LOCATION_CODE_MIN_LENGTH) ||
+        ((i_unexpandedLocationCode.compare(0, 4, "Ufcs") !=
+          constants::STR_CMP_SUCCESS) &&
+         (i_unexpandedLocationCode.compare(0, 4, "Umts") !=
+          constants::STR_CMP_SUCCESS)) ||
+        ((i_unexpandedLocationCode.length() >
+          constants::UNEXP_LOCATION_CODE_MIN_LENGTH) &&
+         (i_unexpandedLocationCode.find("-") != 4)))
```

Comment:
This check is necessary because the length of the unexpanded location code can exceed 4 characters. For instance, consider "Ufcst" - this is an invalid location code. If the given location code is longer than 4 characters, the character following "Ufcs" or "Umts" should be "-". Comparing strings would incorrectly indicate it is a valid location code as the first 4 characters matches.

By performing this check, I can immediately determine that it is an invalid location code without needing to search through the entire JSON file.

Reply:
Ok with Anupama.